FLOE-119

jobara opened the pull request:
Adds a simple function for writing out meta tags and setting itemscope. The markup viewer makes use of this to write out metadata when it replaces the video placeholder markup.

http://issues.fluidproject.org/browse/FLOE-119

jobara committed:
FLOE-119: Utility function for writing metadata

Added a utility function for writing out metadata to an element.

jobara committed:
FLOE-119: Updated the markupViewer

The markup viewer should replace the videoPlaceholder with proper video markup.

jobara committed:
Merge branch 'master' into FLOE-119

jobara committed:
FLOE-119: Bug fixes.

Fixed some issues with the video placeholder replacement.

jobara committed:
Merge branch 'master' into FLOE-119

jobara committed:
FLOE-119: removing captions from meta tag output

Since the captions are handled in a second call to fluid.metadata.writer. I've removed the captions metadata from the initial call to the metadata.writer.

jobara committed:
FLOE-119: Added an all-tests.html file

michelled commented on tests/js/metadataUtilsTests.js:
```diff
@@ -0,0 +1,67 @@
+(function ($) {
+    fluid.registerNamespace("fluid.tests");
+
+    jqUnit.test("fluid.metadata.writer: no type", function () {
```
There seems to be a lot of repetition in these tests - can you try to remove some of the duplication?

michelled commented on demos/js/markupViewer.js:
```diff
@@ -22,34 +22,93 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                src: metadata.url,
+                type: "video/" + metadata.url.split(".").pop()
+            }).appendTo(realMarkup);
+            fluid.metadata.writer(realMarkup, "http://schema.org/Movie", fluid.markupViewer.transformToVideoMetadata(metadata));
```
Can you pull this hardcoded url out somewhere?

jobara committed:
FLOE-119: refactored metadata utils tests.

jobara committed:
FLOE-119: Making a constant for itemtype

So are only have one for MOVIE but more can/will be added as needed.

jobara committed:
FLOE-119: Updating metadata writer

Updated metadata writer based on correct specification for writing metadata. Including the ability to set itemprop on the container.

jobara committed:
FLOE-119: update markup viewer

Updated the markup viewer's replacement markup for the video placeholder with the corrected format for the metadata output. Specifically wrapping the video element in a div and placing the metadata there instead.

jobara commented:
@michelled I've addressed the issues you raised in your code review and updated based on @acheetham comments.

jobara committed:
Merge branch 'master' into FLOE-119

jobara committed:
FLOE-119: Updated button to "View Output HTML"

jobara committed:
FLOE-119: Added comment to transformation function

michelled commented on demos/js/markupViewer.js:
```diff
@@ -22,34 +22,99 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        return videoMetatdata;
+    };
+
+    fluid.markupViewer.replaceVideoPlaceholder = function (content, placeholderID, metadata) {
```
I think it would be better to separate the markup generation from the rest of the work here so that it is easier for someone to customize what gets created.

jobara committed:
FLOE-119: Adding comments

jobara committed:
FLOE-119: Broke out video source generation

video source generation is now in a separate function.

jobara committed:
FLOE-119: Moved some markup to the options.

Moved the videoContainer and captionsContainer markup to be options.

michelled commented:
Merged into project repo at 5faf68d0c06c3541bd21910018d1450f280a50c1
